src/context.rs

```diff
@@ -15,6 +16,26 @@ pub struct Context {
     pub authorization: Option<Authorization>,
     /// Raw authentication data, for use in making HTTP requests as a client.
     pub auth_data: Option<AuthData>,
+    logger: Option<slog::Logger>,
+}
+
+/// Trait for retrieving a logger from a struct.
+pub trait HasLogger {
```

Comment:
This feels like a common pattern - is there a standard naming convention for such traits (and is `Has...` it)?  Maybe worth asking @bossmc?

Reply:
Discussed with Jon Lawn and Mark Thebridge in #rust on slack. Apparently, no clear conventions exist yet. https://github.com/rust-lang-nursery/api-guidelines/issues/28

In the absence of such I'm quite happy with HasLogger for now. Logger is potentially confusing, since it sounds like context could then be used to log.

That being said, there are conventions for getter and setter methods, and it looks like get_logger() should just be logger(), so changing that.

Reply:
Agree that there's no strong convention yet.

A more generic solution (though maybe too java-y?) would be:

```
trait Contain<T> {
    get(&self) -> &T;
    set(&mut self, T);
}

impl Contain<&Option<Logger>> for Context {
    ...
}
```